generators/python/src/fern_python/generators/pydantic_model/fern_aware_pydantic_model.py

```diff
@@ -201,16 +201,23 @@ def _type_id_for_forward_ref(self) -> Optional[ir_types.TypeId]:
 
     def _must_import_after_current_declaration(self, type_name: ir_types.DeclaredTypeName) -> bool:
         type_id_to_reference = self._type_id_for_forward_ref()
-        should_import_after = False
-        if type_id_to_reference is not None:
-            should_import_after = self._context.does_type_reference_other_type(
-                type_id=type_name.type_id, other_type_id=type_id_to_reference
-            )
+        if type_id_to_reference is None:
+            return False
 
-        if should_import_after:
+        if self._context.does_type_reference_other_type(type_id=type_name.type_id, other_type_id=type_id_to_reference):
+            # direct forward reference
             self._model_contains_forward_refs = True
+            return True
+
+        for referenced_type_id in self._context.get_referenced_types(type_id_to_reference):
+            if self._context.does_type_reference_other_type(
+                type_id=referenced_type_id, other_type_id=type_name.type_id
+            ):
+                # indirect forward reference, i.e. through a container (e.g. list, optional, etc.)
```

Comment:
Is this _always_ a container? From the implementation, it sounds like this could just be any of the other referenced types -- might these always just be named types instead?

Reply:
Discussed offline - we might actually look deeper into the `does_type_reference_other_type` function. I have a feeling we aren't visiting the nodes as we expect there.

Reply:
Done. Much cleaner.

Reply:
Went ahead and took a different approach instead, looking at if the thing we're importing is part of a circular dependency chain with itself.